source/Octopus.Cli/Commands/WorkerPool/CleanWorkerPoolCommand.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Octopus.Cli.Infrastructure;
+using Octopus.Cli.Repositories;
+using Octopus.Cli.Util;
+using Octopus.Client;
+using Octopus.Client.Model;
+using Octopus.Client.Model.Endpoints;
+
+namespace Octopus.Cli.Commands.WorkerPools
+{
+    [Command("clean-workerpool", Description = "Cleans all Offline Worker Machines from a WorkerPool")]
+    public class CleanWorkerPoolCommand : ApiCommand, ISupportFormattedOutput
+    {
+        string poolName;
+        readonly HashSet<string> statuses = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        readonly HashSet<string> healthStatuses = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        private bool? isDisabled;
+        private bool? isCalamariOutdated;
+        private bool? isTentacleOutdated;
+        WorkerPoolResource workerpoolResource;
+        IEnumerable<WorkerMachineResource> machines;
+        List<MachineResult> commandResults = new List<MachineResult>();
+
+
+        public CleanWorkerPoolCommand(IOctopusAsyncRepositoryFactory repositoryFactory, IOctopusFileSystem fileSystem, IOctopusClientFactory clientFactory, ICommandOutputProvider commandOutputProvider)
+            : base(clientFactory, repositoryFactory, fileSystem, commandOutputProvider)
+        {
+            var options = Options.For("WorkerPool Cleanup");
+            options.Add("workerpool=", "Name of a worker pool to clean up.", v => poolName = v);
+            options.Add("status=", $"Status of Worker Machines to clean up ({string.Join(", ", HealthStatusProvider.StatusNames)}). Can be specified many times.", v => statuses.Add(v));
+            options.Add("health-status=", $"Health status of Worker Machines to clean up ({string.Join(", ", HealthStatusProvider.HealthStatusNames)}). Can be specified many times.", v => healthStatuses.Add(v));
+            options.Add("disabled=", "[Optional] Disabled status filter of Worker Machine to clean up.", v => SetFlagState(v, ref isDisabled));
+            options.Add("calamari-outdated=", "[Optional] State of Calamari to clean up. By default ignores Calamari state.", v => SetFlagState(v, ref isCalamariOutdated));
+            options.Add("tentacle-outdated=", "[Optional] State of Tentacle version to clean up. By default ignores Tentacle state", v => SetFlagState(v, ref isTentacleOutdated));
+        }
+
+        public async Task Request()
+        {
+            if (string.IsNullOrWhiteSpace(poolName))
+                throw new CommandException("Please specify a worker pool name using the parameter: --workerpool=XYZ");
+            if (!healthStatuses.Any() && !statuses.Any())
+                throw new CommandException("Please specify a status using the parameter: --health-status");
```

Comment:
Should this mention the `status` parameter as you can use either or both the `health-status` and `status` parameters to filter the machines.

Reply:
I interpreted this https://github.com/OctopusDeploy/OctopusClients/blob/master/source/Octopus.Cli/Commands/HealthStatusProvider.cs#L13-L16 as meaning that the `status` was for backward compatibility.

I thought cause worker pools is new, it doesn't need the backward compatibility ??

Reply:
ooohh, whoops, if I meant what I said I would have removed the status option.

Reply:
ok, I'm just confused, will redo.